flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointStorageAccess.java

```diff
@@ -110,9 +110,11 @@ public boolean supportsHighlyAvailableStorage() {
 
     @Override
     public void initializeBaseLocations() throws IOException {
-        fileSystem.mkdirs(sharedStateDirectory);
-        fileSystem.mkdirs(taskOwnedStateDirectory);
-        baseLocationsInitialized = true;
+        if (!baseLocationsInitialized) {
```

Comment:
Why we must change this?

Reply:
This is unnecessary. I thought the savepoint will use the directory before, then I realized there's a variable `defaultSavepointDirectory` for savepoints.